fix Windows build warning

When we build modmesh on the WIndows platform, we can get warning message as follows:

```
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\include\xutility(254,82): warning C4267: 'argument': conversion from 'size_t' to 'modmesh::StaticMesh::uint_type', possible loss of data [D:\a\modmesh\modmesh\build\cpp\modmesh\modmesh_primary.vcxproj]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\include\memory(2104,18): message : see reference to function template instantiation 'void std::_Construct_in_place<_Ty,unsigned int,size_t&,int,unsigned __int64,modmesh::StaticMesh::ctor_passkey>(_Ty &,unsigned int &&,size_t &,int &&,unsigned __int64 &&,modmesh::StaticMesh::ctor_passkey &&) noexcept(false)' being compiled [D:\a\modmesh\modmesh\build\cpp\modmesh\modmesh_primary.vcxproj]
          with
          [
              _Ty=modmesh::StaticMesh
          ]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\include\memory(2779,46): message : see reference to function template instantiation 'std::_Ref_count_obj2<modmesh::StaticMesh>::_Ref_count_obj2<_Ty,size_t&,int,unsigned __int64,modmesh::StaticMesh::ctor_passkey>(_Ty &&,size_t &,int &&,unsigned __int64 &&,modmesh::StaticMesh::ctor_passkey &&)' being compiled [D:\a\modmesh\modmesh\build\cpp\modmesh\modmesh_primary.vcxproj]
          with
          [
              _Ty=unsigned int
          ]
D:\a\modmesh\modmesh\cpp\modmesh/mesh/StaticMesh.hpp(264,21): message : see reference to function template instantiation 'std::shared_ptr<_Ty> std::make_shared<modmesh::StaticMesh,unsigned int,size_t&,int,unsigned __int64,modmesh::StaticMesh::ctor_passkey>(unsigned int &&,size_t &,int &&,unsigned __int64 &&,modmesh::StaticMesh::ctor_passkey &&)' being compiled [D:\a\modmesh\modmesh\build\cpp\modmesh\modmesh_primary.vcxproj]
          with
          [
              _Ty=modmesh::StaticMesh
          ]
D:\a\modmesh\modmesh\cpp\modmesh\inout\gmsh.cpp(72,53): message : see reference to function template instantiation 'std::shared_ptr<_Ty> modmesh::StaticMesh::construct<unsigned int,size_t&,int,unsigned __int64>(unsigned int &&,size_t &,int &&,unsigned __int64 &&)' being compiled [D:\a\modmesh\modmesh\build\cpp\modmesh\modmesh_primary.vcxproj]
          with
          [
              _Ty=modmesh::StaticMesh
          ]
```

we may want to address these warnings to have a cleaner build.

--- a/cpp/modmesh/inout/gmsh.cpp
+++ b/cpp/modmesh/inout/gmsh.cpp
@@ -69,7 +69,11 @@ Gmsh::Gmsh(const std::string & data)
 
 std::shared_ptr<StaticMesh> Gmsh::to_block()
 {
-    std::shared_ptr<StaticMesh> block = StaticMesh::construct(m_eldim.max(), m_nds.shape(0), 0, m_elems.size());
+    std::shared_ptr<StaticMesh> block = StaticMesh::construct(
+        m_eldim.max(),
+        static_cast<StaticMesh::uint_type>(m_nds.shape(0)),
+        0,
+        static_cast<StaticMesh::uint_type>(m_elems.size()));
     build_interior(block);
     return block;
 }

